tests/phpunit/integration/Modules/Analytics_4/Report/Custom_Dimensions_Response_ParserTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Class Google\Site_Kit\Tests\Modules\Analytics_4\Report\Custom_Dimensions_Response_ParserTest
+ *
+ * @package   Google\Site_Kit\Tests\Modules\Analytics_4\Report
+ * @copyright 2023 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Modules\Analytics_4\Report;
+
+use Google\Site_Kit\Tests\TestCase;
+use Google\Site_Kit\Modules\Analytics_4\Report\Custom_Dimensions_Response_Parser;
+use Google\Site_Kit_Dependencies\Google\Service\AnalyticsData\RunReportResponse as Google_Service_AnalyticsData_RunReportResponse;
+use Google\Site_Kit_Dependencies\Google\Service\AnalyticsData\DimensionHeader as Google_Service_AnalyticsData_DimensionHeader;
+use Google\Site_Kit_Dependencies\Google\Service\AnalyticsData\DimensionValue as Google_Service_AnalyticsData_DimensionValue;
+use Google\Site_Kit_Dependencies\Google\Service\AnalyticsData\Row as Google_Service_AnalyticsData_Row;
+
+/**
+ * @group Modules
+ * @group Analytics_4
+ * @group Report
+ */
+class Custom_Dimensions_Response_ParserTest extends TestCase {
+
+	protected function create_rows_with_dimension_values( $values ) {
+		$rows = array();
+		foreach ( $values as $key => $value ) {
+			$dimension_value = new Google_Service_AnalyticsData_DimensionValue();
+			$dimension_value->setValue( $value );
+			$rows[ $key ] = new Google_Service_AnalyticsData_Row();
+			$rows[ $key ]->setDimensionValues( array( $dimension_value ) );
+		}
+		return $rows;
+	}
+
+	public function test_swap_custom_dimension_ids_with_names__post_author() {
+		$response                     = new Google_Service_AnalyticsData_RunReportResponse();
+		$dimension_header_post_author = new Google_Service_AnalyticsData_DimensionHeader();
+		$dimension_header_post_author->setName( 'customEvent:googlesitekit_post_author' );
+		$response->setDimensionHeaders( array( $dimension_header_post_author ) );
+
+		$rows = array();
+
+		// Existing author with a valid display name.
+		$author_id = $this->factory()->user->create( array( 'display_name' => 'test author 1' ) );
+
+		// 5000 would be a non-existent user and `(not set)` is an invalid user_id.
+		$rows = $this->create_rows_with_dimension_values( array( (string) $author_id, '5000', '(not set)' ) );
+
+		$response->setRows( $rows );
+		$custom_dimension_query = new Custom_Dimensions_Response_Parser( $response );
+		$custom_dimension_query->swap_custom_dimension_ids_with_names();
+
+		$swappedRows = $response->getRows();
```

Comment:
This should be the same as `$rows` since only their internal properties are changing. The instances are the same.